### PR TITLE
upgrading list

### LINKED ISSUE #1259: 2.0 - Update List
This is a tracking issue for updating List

**Scope:** List

### LINKED ISSUE #1293: Standard API for major event handlers in 2.0
While we're writing components for 2.0, we should standardize the API got event handlers.

Currently with props all event handlers are native event handlers.
E.g: `onChange` of a checkbox is the native onChange of `input type=checkbox`
This leads to some problems, mdc-web introduce some additional properties on top of native controls.
E.g. checkbox has a `indeterminate` state, select has `selectedIndex` etc.

Getting these values from native event handlers which just pass the `Event`, is tricky and most times requires creating a ref.

In order to solve this we can send added details to major event handlers(not necessarily all) like onChange for checkbox, select, onClick for button etc.

Current TS definition:
```
onClick: (e:Event) => void;
```

Updated version:

```
onChange:(e:Event, object: {
  MDComponent: MaterialComponent;
  checked: boolean;
  indeterminate: boolean;
}) => void
```

The additional object makes sure that it is not counter intuitive to users if 1.0

### PR DESCRIPTION
**Pull request by prateekbh**
- Fixes #1259 
- confirms to #1293

### CODE REVIEW

File: packages/list/package.json [hunk 1]
```diff
@@ -31,13 +31,13 @@
     "url": "https://github.com/prateekbh/preact-material-components/issues"
   },
   "dependencies": {
-    "@material/list": "0.39.3",
+    "@material/list": "1.1.1",
```
Comment (cromefire) [inline packages/list/package.json]: Do all components have the same version? I saw some with `1.1.0`

**Comment by cromefire** [comment]
I hope I have some time in the next few day, so I can help some more

File: packages/list/package.json (hunk 1 above)
Comment (prateekbh) [reply packages/list/package.json]: There might have been a release recently we'll upgrade all in the end once

Comment (cromefire) [reply packages/list/package.json]: That's fine

**Pull request merged**